documentation/client-side-forms/appendix-vaadin-components.asciidoc

```diff
@@ -0,0 +1,284 @@
+    `)))}
+  </vaadin-radio-group>
+
+  <vaadin-list-box ...="${field(this.binder.model.mySingleSelectionField)}" label="list-box">
+    ${until(endpoint.getMyOptions().then(opts => repeat(opts, (item) => html`
+      <vaadin-item><span>${item}</span></vaadin-item>
+    `)))}
+  </vaadin-list-box>
+  `;
+}
+----
+
+==== Single selection using index
+
+To select items by index, the `vaadin-select` component should be used. It accepts an integer for the index value.
+Because this component is configurable via the `template` tag, it's not possible to set the options with an asynchronous call.
+
+[source, typescript]
+----
+import '@vaadin/vaadin-select';
+...
+render() {
+  return html`
+  <vaadin-select ...="${field(this.binder.model.mySelectField)}" label="select">
+    <template>
+      <vaadin-list-box>
+        <vaadin-item><span>item-1</span></vaadin-item>
+        <vaadin-item><span>item-2</span></vaadin-item>
+      </vaadin-list-box>
+    </template>
+  </vaadin-select>
+  `;
+}
+----
+
+==== Multiple selection
+
+The Vaadin component for multiple selection is the `vaadin-checkbox-group` which accepts an array of strings.
+
+[source, typescript]
+----
+import '@vaadin/vaadin-checkbox/vaadin-checkbox-group';
+...
+render() {
+  return html`
+  <vaadin-checkbox-group ...="${field(this.binder.model.myListField)}" label="check-group">
+    ${until(endpoint.getMyOptions().then(opts => repeat(opts, (item) => html`
+      <vaadin-checkbox value="${item}">${item}</vaadin-checkbox>
+    `)))}
+  </vaadin-checkbox-group>
+  `;
+}
+----
+
+
+=== Binding data to Date and Time fields
+
+Use `vaadin-date-picker` for binding to Java `LocalDate`, `vaadin-time-picker` for `LocalTime`, and `vaadin-date-time-picker` for `LocalDateTime`.
+
+[source, typescript]
+----
+import '@vaadin/vaadin-date-picker';
+import '@vaadin/vaadin-time-picker';
+import '@vaadin/vaadin-date-time-picker';
+...
+render() {
+  return html`
+  <vaadin-date-picker ...="${field(this.binder.model.myDateField)}" label="date"></vaadin-date-picker>
+  <vaadin-time-picker ...="${field(this.binder.model.myTimeField)}" label="time"></vaadin-time-picker>
+  <vaadin-date-time-picker ...="${field(this.binder.model.myDateTimeField)}" label="date-time"></vaadin-date-time-picker>
+  `;
+}
+----
+
+=== Wrapping components in custom fields
+
+Vaadin provides the `vaadin-custom-field` that can be used to wrap one or multiple vaadin fields.
+It works with the following components:
+
+  - `vaadin-text-field`
+  - `vaadin-number-field`
+  - `vaadin-password-field`
+  - `vaadin-text-area`
+  - `vaadin-select`
+  - `vaadin-combo-box`
+  - `vaadin-date-picker`
+  - `vaadin-time-picker`
+
+[source, typescript]
+----
+import '@vaadin/vaadin-custom-field';
+import '@vaadin/vaadin-text-field';
+...
+render() {
+  return html`
+  <vaadin-custom-field ...="${field(this.binder.model.myTextField)}" label="custom-field">
+    <vaadin-text-field></vaadin-text-field>
+  </vaadin-custom-field>
```

Comment:
Not a realistic use case to wrap only a `vaadin-text-field` in a custom field. Imo it would be better to have an example here where at least 2 components have been wrapped into a custom field.